include/dali/core/mm/detail/free_list.h

```diff
@@ -213,7 +213,8 @@ class best_fit_free_list {
    * @brief Places the free block in the list.
    *
    * The block is not merged with adjacent blocks. The fragmentation of the list is permanent.
-   * For long term usage, when fragmentation is an issue, use coalescing_free_list or free_tree.
+   * For long term usage, when internal fragmentation is an issue, use coalescing_free_list or
+   * coalescing_free_tree.
```

Comment:
Coalescing free lists help with internal fragmentation, but may actually aggravate external fragmentation.

Reply:
Maybe you should add this in the comment as well?